src/scheduler/MpiWorld.cpp

```diff
@@ -345,23 +413,14 @@ int MpiWorld::isend(int sendRank,
                     faabric::MPIMessage::MPIMessageType messageType)
 {
     int requestId = (int)faabric::util::generateGid();
+    auto it = reqIdToRanks.try_emplace(
+      requestId, MPI_IS_ISEND_REQUEST, MPI_IS_ISEND_REQUEST);
+    if (!it.second) {
+        SPDLOG_ERROR("Request ID {} is already present in map", requestId);
+        throw std::runtime_error("Request ID already in map");
+    }
```

Comment:
I don't understand what this check is guarding against, surely this would only trigger if `generateGid` wasn't working properly? We've talked about `try_emplace` before and I think this form of check is quite verbose, so we should use it sparingly. When we do, we should add a comment as to why.

Reply:
You are right. As we discussed `try_emplace` and `emplace` are mostly interchangable. The main problem here is that what I check would only trigger if `generateGid` wasn't working properly, and this is not the right place to check it.